### PR TITLE
Put create snow cluster under feature gate

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/eksctl-anywhere/cmd/createcluster.go [hunk 1]
```diff
@@ -110,28 +110,32 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command) error {
...
-	if deps.Provider.Name() == "tinkerbell" {
+	if deps.Provider.Name() == constants.TinkerbellProviderName {
 		flag := cmd.Flags().Lookup("hardwarefile")
 		if flag == nil {
-			return fmt.Errorf("Something wrong. Flag hardwarefile not set up for provider tinkerbell")
+			return fmt.Errorf("something wrong. Flag hardwarefile not set up for provider tinkerbell")
 		}
 		if !viper.IsSet("hardwarefile") || viper.GetString("hardwarefile") == "" {
-			return fmt.Errorf("Error: required flag \"hardwarefile\" not set")
+			return fmt.Errorf("srror: required flag \"hardwarefile\" not set")
```
Comment (mitalipaygude) [inline cmd/eksctl-anywhere/cmd/createcluster.go:123]: typo?

Comment (mitalipaygude) [reply cmd/eksctl-anywhere/cmd/createcluster.go:123]: Same for all the error statements below

Comment (jiayiwang7) [reply cmd/eksctl-anywhere/cmd/createcluster.go:123]: damn.. good catch

Comment (jiayiwang7) [reply cmd/eksctl-anywhere/cmd/createcluster.go]: updated

**Pull request merged**